components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -1206,6 +1206,108 @@ public void removeSubscription(APIIdentifier identifier, int applicationId)
 		}
 	}
 
+	/**
+	 * Moves subscriptions of one app to another app
+	 *
+	 * @param fromIdentifier subscriptions of this app
+	 * @param toIdentifier   will be moved into this app
+	 * @return number of subscriptions moved
+	 * @throws AppManagementException
+	 */
+	public int moveSubscriptions(APIIdentifier fromIdentifier, APIIdentifier toIdentifier)
+			throws AppManagementException {
+
+		Connection conn = null;
+		ResultSet results = null;
+		PreparedStatement ps = null;
+		int fromAppId = -1;
+		int toAppId = -1;
+		int count;
+
+		try {
+
+			String getAppIdQuery = "SELECT APP_ID FROM APM_APP " +
+					"WHERE APP_PROVIDER=? AND APP_NAME=? AND APP_VERSION=?";
+
+			try {
+				conn = APIMgtDBUtil.getConnection();
+				ps = conn.prepareStatement(getAppIdQuery);
+			} catch (SQLException e) {
+				handleException("Failed to create database connection ", e);
+			}
+
+			try {
+				ps.setString(1, fromIdentifier.getProviderName());
+				ps.setString(2, fromIdentifier.getApiName());
+				ps.setString(3, fromIdentifier.getVersion());
+				results = ps.executeQuery();
+				if (results.next()) {
+					fromAppId = results.getInt(1);
+				}
+			} catch (SQLException e) {
+				APIMgtDBUtil.closeAllConnections(ps, conn, results);
+				String msg = "Could not retrieve app ID of " + fromIdentifier.getProviderName() +
+						"-" + fromIdentifier.getApiName() + "-" + fromIdentifier.getVersion();
+				handleException(msg, e);
+			}
+			APIMgtDBUtil.closeAllConnections(null, null, results);
+			if (fromAppId == -1) {
+				if (log.isDebugEnabled()) {
+					log.debug("Could not find app ID of 'from' app " + fromIdentifier);
+				}
+				return -1;
+			}
+
+			try {
+				ps.setString(1, toIdentifier.getProviderName());
+				ps.setString(2, toIdentifier.getApiName());
+				ps.setString(3, toIdentifier.getVersion());
+				results = ps.executeQuery();
+				if (results.next()) {
+					toAppId = results.getInt(1);
+				}
+			} catch (SQLException e) {
+				APIMgtDBUtil.closeAllConnections(ps, conn, results);
+				String msg = "Could not retrieve app ID of " + toIdentifier.getProviderName() + "-"
+						+ toIdentifier.getApiName() + "-" + toIdentifier.getVersion();
+				handleException(msg, e);
+			}
+			APIMgtDBUtil.closeAllConnections(null, null, results);
+			if (toAppId == -1) {
+				if (log.isDebugEnabled()) {
+					log.debug("Could not find app ID of 'to' app " + toIdentifier);
+				}
+				return -1;
+			}
+
+			String moveQuery = "UPDATE APM_SUBSCRIPTION SET APP_ID=? WHERE APP_ID=?";
+			count = -1;
```

Comment:
Now the `moveSubscriptions` method is broken into to methods, see commit 07d4afd84e007233c5a9ee791d8288c5bffb3cbf of PR #153